glib/src/object.rs

```diff
@@ -2199,6 +2200,22 @@ pub trait ObjectExt: ObjectType {
     /// Downgrade this object to a weak reference.
     fn downgrade(&self) -> WeakRef<Self>;
 
+    // rustdoc-stripper-ignore-next
+    /// Add a callback to be notified when the Object is disposed.
+    #[doc(alias = "g_object_weak_ref")]
+    #[doc(alias = "connect_drop")]
+    fn add_weak_ref_notify<F: FnOnce() + Send + Sync + 'static>(&self, f: F)
```

Comment:
@bilelmoussaoui @GuillaumeGomez Do you maybe have a better suggestion for the name? The C API is called `g_object_weak_ref()` but that's even worse :)

Reply:
`add_weak_ref_callback`?

Reply:
The current one is more descriptive than that: notification vs. callback is clearer